src/perception/src/utils/cluster.cpp

```diff
@@ -0,0 +1,37 @@
+#include <utils/cluster.hpp>
+
+Cluster::Cluster(pcl::PointCloud<pcl::PointXYZI>::Ptr point_cloud) : point_cloud(point_cloud) {
+    color = "undefined";
+    centroidIsDefined = false;
+}
+
+
+Eigen::Vector4f Cluster::getCentroid() {
+    if (centroidIsDefined) return this->centroid;
+
+    Eigen::Vector4f tempCentroid;
+    pcl::compute3DCentroid(*point_cloud, centroid);
+    this->centroidIsDefined = true;
+
+    return centroid;
+}
+
+std::string Cluster::getColor() {
+    return color;
+}
+
+void Cluster::setColor(std::string new_color) {
+    this->color = new_color;
```

Comment:
As you are using the color as a string (instead of a enum maybe), you should probably impede the setting of the color to a value that is not one of the ones allowed